src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -72,11 +79,17 @@ module MOM_ice_shelf
 type, public :: ice_shelf_CS ; private
   ! Parameters
   type(MOM_restart_CS), pointer :: restart_CSp => NULL() !< A pointer to the restart control
-                                          !! structure for the ice shelves
-  type(ocean_grid_type) :: grid           !< Grid for the ice-shelf model
+                                                  !! structure for the ice shelves
+  type(ocean_grid_type), pointer :: Grid_in => NULL()    !< un-rotated input grid metric
+  type(hor_index_type), pointer :: HI_in => NULL()  !< Pointer to a horizontal indexing structure for
+                                                    !! incoming data which has not been rotated.
+  type(hor_index_type), pointer :: HI => NULL()  !< Pointer to a horizontal indexing structure for
+                                                 !! incoming data which has not been rotated.
+  logical :: rotate_index = .false.   !< True if index map is rotated
+  integer :: turns                    !< The number of quarter turns for rotation testing.
+  type(ocean_grid_type), pointer :: Grid => NULL() !< Grid for the ice-shelf model
```

Comment:
Based on the `MOM_CS` implementation, I have the following comments:
1. `Grid_in` should not need to be a pointer.  `Grid` is a pointer because it could either point to `Grid_in` or to its own (anonymous) rotated value, but `Grid_in` does not have that constraint.
2. This is named `G` in MOM, should we follow the same convention here?
3. `HI` and `HI_in` are probably not needed, since they would generally be stored in `Grid`.
4. Similarly, `turns` is stored in `Grid%HI%turns` so probably does not need to be added.

But if there's reasons for introducing these separately then disregard this comment.